cli/client.go

```diff
@@ -639,7 +754,17 @@ func isProdBaseURL(baseURL *url.URL) bool {
 	return strings.HasSuffix(baseURL.Hostname(), "viam.com")
 }
 
+type ctxKey int
+
+const ctxKeyViamClientForTesting ctxKey = iota
+
 func newViamClient(c *cli.Context) (*viamClient, error) {
+	// quick hack to avoid dealing with any of the logic below. some testing
+	// may want to use this for integration testing.
+	if val := c.Context.Value(ctxKeyViamClientForTesting); val != nil {
```

Comment:
We've been avoiding this logic by having something like `MachinesPartCopyFilesAction` call `newViamClient` and _then_ call an unexported `machinesPartCopyFilesAction`. Then, the tests can call `machinesPartCopyFilesAction` and skip `viamClient` creation. Did you have another reason for using this pattern? Otherwise, I wonder if we could use the existing pattern.

Reply:
Nah I was just copying and pasting. I'll change it